### PR TITLE
Updates license headers

### LINKED ISSUE #1744: [BUG] Incorrect license headers
**Is your feature request related to a problem?**
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/138, some obsolete ALv2 headers are still present, e.g. [here](https://github.com/opensearch-project/security/blob/2972bd15a98dbaa8c571b9b302268de076d20ee0/src/test/java/org/opensearch/security/TracingTests.java#L17). 

**What solution would you like?**
1. Add automated license header Gradle checks to CI similar to [this](https://github.com/opensearch-project/asynchronous-search/commit/bc483fe7a253d20216ce5fc4d5427e42819b7aaf).
2. Update the default header for all new files created by IDEs similar to https://github.com/opensearch-project/OpenSearch/pull/510.
3. Fix any incorrect headers.

**Do you have any additional context?**
Licenses must match https://github.com/opensearch-project/opensearch-plugins/blob/main/HEADERS.md

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Updates license headers
* Category : Bug fix

### Issues Resolved
* Resolves #1744

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
@opensearch-project/security team, let's get this reviewed and merged and then we can open a follow-up issue to generate new license headers on new file creation.

**Review by peternied — CHANGES_REQUESTED** [review]
Taking this change without being sure that we correctly updated all of the files makes it hard to know if this is the correct pull request to merge.  I would rather we solved the problem of detecting incorrect license headers first and then rolled out the fix for the improperly documented files.

I believe that OpenSearch already has a [process](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/groovy/org/opensearch/gradle/precommit/PrecommitTasks.groovy#L46) for doing this, can we see if we can adopt it or a similar process?

**Comment by DarshitChanpura** [comment]
> Taking this change without being sure that we correctly updated all of the files makes it hard to know if this is the correct pull request to merge. I would rather we solved the problem of detecting incorrect license headers first and then rolled out the fix for the improperly documented files.
> 
> I believe that OpenSearch already has a [process](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/groovy/org/opensearch/gradle/precommit/PrecommitTasks.groovy#L46) for doing this, can we see if we can adopt it or a similar process?

The license headers are updated according to the step mentioned in the bug attached here. I have replaced all the headers similar to [this](https://github.com/opensearch-project/security/blob/2972bd15a98dbaa8c571b9b302268de076d20ee0/src/test/java/org/opensearch/security/TracingTests.java#L17) in all the files to match the header mentioned [here](https://github.com/opensearch-project/opensearch-plugins/blob/main/HEADERS.md). 

The follow-up to this would to add automated checks i.e. one which you mentioned in the comment, using a groovy task or an alternative in our project which would need more time. 

I did try by adding a task using spotless, but that would replace all the license headers with the new one which is not correct, so ended up removing it and would need an extra set of hands in getting this fix through.

**Comment by peternied** [comment]
> The follow-up to this would to add automated checks i.e. one which you mentioned in the comment, using a groovy task or an alternative in our project which would need more time.

Could we add a check for the bad license header to make sure we don't add any more violations like these back in?  I don't think we need a 'license header corrector' tool

File: config/checkstyle/sun_checks.xml [hunk 1]
```diff
@@ -204,4 +204,12 @@
     <property name="severity" value="warning"/>
   </module>
 
+  <!-- Checks for Headers to not contain old license text -->
+  <module name="RegexpSingleline">
+    <property name="format" value="Portions Copyright OpenSearch Contributors"/>
```
Comment (DarshitChanpura) [inline config/checkstyle/sun_checks.xml:209]: For now I'm checking the starting line of the old license header to see if any files contain it, if so checkstyle will throw violation.

**Comment by DarshitChanpura** [comment]
> > The follow-up to this would to add automated checks i.e. one which you mentioned in the comment, using a groovy task or an alternative in our project which would need more time.
> 
> Could we add a check for the bad license header to make sure we don't add any more violations like these back in? I don't think we need a 'license header corrector' tool

Done

File: config/checkstyle/sun_checks.xml (hunk 1 above)
Comment (peternied) [inline config/checkstyle/sun_checks.xml:209]: Looks like that regex needs a bit of improvment, here 
```suggestion
    <property name="format" value="\*( Portions)? Copyright OpenSearch Contributors"/>
```

With this change I found 5 errors in the following files:
```
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/transport/DualModeSSLHandler.java:2: Incorrect license header [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/transport/SSLConfig.java:2: Incorrect license header [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/util/SSLConnectionTestUtil.java:2: Incorrect license header [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/ssl/util/TLSUtil.java:2: Incorrect license header [RegexpSingleline]
[ant:checkstyle] [ERROR] /local/home/petern/git/security/src/main/java/org/opensearch/security/tools/SecurityAdmin.java:17: Incorrect license header [RegexpSingleline]
```

**Review by peternied — CHANGES_REQUESTED** [review]

File: config/checkstyle/sun_checks.xml (hunk 1 above)
Comment (DarshitChanpura) [reply config/checkstyle/sun_checks.xml:209]: good catch... thank you!

**Comment by peternied** [comment]
> should we update NOTICE.txt?

I don't think we should modify the NOTICE.txt at this time.

**Review by peternied — APPROVED** [review]

**Review by cliu123 — APPROVED** [review]

**Pull request merged**